src/extraction/index.js

```diff
@@ -28,40 +28,72 @@ const EXTRACTION_METHODS = {
   open_graph: extractProductWithOpenGraph,
 };
 
+const IS_BACKGROUND_UPDATE = (function isBackgroundUpdate() {
```

Comment:
Fun fact, since we use `let` and `const` now we don't need to use functions to hide one-time computations, blocks work just as well:

```js
let isBackgroundUpdate = false;
{ // This is a block! They are neat!
  let result = false;
  try {
    isBackgroundUpdate = window.top.location.href.startsWith(
      browser.runtime.getURL('/'), // URL is unique per-install / hard to forge
    );
  } catch (err) {
    // Non-background updates may throw a cross-origin error
  }
}
```

although since we don't need the intermediate result value anyway we can simplify even more:

```js
let isBackgroundUpdate = false;
try {
  isBackgroundUpdate = window.top.location.href.startsWith(
    browser.runtime.getURL('/'), // URL is unique per-install / hard to forge
  );
} catch (err) {
  // Non-background updates may throw a cross-origin error
}
```

Reply:
Yeah I haven't really made use of blocks, but I know what you mean in that `const` and `let` are block-scoped. Makes sense.